nodejs/awsx/cloudwatch/widgets_json.ts

```diff
@@ -21,14 +21,19 @@ import { YAxis } from "./widgets_graph";
 // the json conversion properly as it's very easy to screw up if using things like the 'any' type.
 
 export interface WidgetJson {
-    type: pulumi.Input<"metric" | "text">;
+    type: pulumi.Input<"alarm" | "metric" | "text">;
     x: pulumi.Input<number>;
     y: pulumi.Input<number>;
     width: pulumi.Input<number>;
     height: pulumi.Input<number>;
     properties: Record<string, any>;
 }
 
+export interface AlarmWidgetJson extends WidgetJson {
+  type: pulumi.Input<"alarm">;
+  properties: pulumi.Input<{ alarms: pulumi.Input<string>[] }>;

```

Comment:
Looking at https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/CloudWatch-Dashboard-Body-Structure.html#CloudWatch-Dashboard-Properties-Alarm-Widget-Object -- it seems there are a number of other keys that can actually exist under `properties`. Do you mind extending this to be similar to `MetricWidgetJson` where you defined an `AlarmWidgetPropertiesJson` and include those additional fields (as optional)?

Reply:
I've pushed an update, wasn't sure if I should be breaking out the values of `states` and `sortBy` further.  Also, should I squash my commits?

Please advise.

Reply:
> I've pushed an update, wasn't sure if I should be breaking out the values of states and sortBy further.

Do you mean into their own types? I think it's fine to inline as you have it for now.

Reply:
Yes, that's what I meant, they could be enums.